users/userRouter.js

```diff
@@ -1,47 +1,140 @@
 
-const router = express.Router();
+const Users = require("./userDb.js");
 
-router.post('/', (req, res) => {
+const router = express.Router();
 
+router.post("/", validateUser, (req, res) => {
+  const user = req.body;
+  Users.insert(user)
+    .then(user => {
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "cannot post this user idk"
+      });
+    });
 });
 
-router.post('/:id/posts', (req, res) => {
-
+router.post("/:id/posts", validateUserId, validatePost, (req, res) => {
+  const user = req.body;
+  Users.insert(user)
+    .then(user => {
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "what do i even put here?"
+      });
+    });
 });
 
-router.get('/', (req, res) => {
-
+router.get("/", (req, res) => {
+  Users.get()
+    .then(users => {
+      res.status(200).json(users);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "am i getting it?"
+      });
+    });
 });
 
-router.get('/:id', (req, res) => {
-
+router.get("/:id", validateUserId, (req, res) => {
+  const user = req.user;
+  res.status(200).json(user);
 });
 
-router.get('/:id/posts', (req, res) => {
-
+router.get("/:id/posts", validateUserId, validatePost, (req, res) => {
+  const { id } = req.params;
+
+  Users.getUserPosts(id)
+    .then(posts => {
+      res.status(200).json(posts);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({
+        message: "could not get it."
+      });
+    });
 });
 
-router.delete('/:id', (req, res) => {
-
+router.delete("/:id", validateUserId, (req, res) => {
+  const { id } = req.params;
+
+  Users.remove(id)
+    .then(id => {
+      res.status(200).json(posts);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({
+        message: "couldn't delete that user"
+      });
+    });
 });
 
-router.put('/:id', (req, res) => {
-
+router.put("/:id", validateUserId, (req, res) => {
+  const { id } = req.params;
+  const addUser = req.body;
+
+  Users.update(id, addUser)
+    .then(addUser => {
+      res.status(200).json(addUser);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "unabele to update the user"
+      });
+    });
 });
 
 //custom middleware
 
 function validateUserId(req, res, next) {
-
-};
+  const { id } = req.params;
+
+  Users.getById(id)
+    .then(user => {
+      // console.log(user);
+      if (!user) {
+        res.status(400).json({
+          message: "invalid user id"
+        });
+      } else {
+        req.user = user;
+        next();
+      }
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ message: "ammmmmm it failed" });
+    });
+}
 
 function validateUser(req, res, next) {
-
-};
+  // idk writing something random copying logic from above.
+  if (!req.body) {
+    res.status(400).json({ message: "missing user data" });
```

Comment:
`!req.body` isn't really going to work here. Even if no body is sent with the request it will still be an empty object. Realistically we should be checking for req.body.name. In an actual app you would probably be checking for name, email, password, maybe more.